artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -828,6 +834,11 @@ public void removeConsumer(final Consumer consumer) {
             }
          }
 
+         if (consumerList.isEmpty()) {
+            // reset when all consumers are removed
+            supportsDirectDeliver = true;
```

Comment:
should maybe iterate the consumers list and set to true if all consumers support direct? e.g. was 3 consumers, 2 core and 1 amqp, then the amqp one removes, leaving 2 core would you want it to go back to direct.

e.g. (pseudo)
```
boolean supportsDirectDeliver = true;
for (consumer : consumerList) {
   if (!consumer.supportsDirectDelivery()){
      supportsDirectDeliver = false;
   } 
}
this.supportsDirectDeliver = supportsDirectDeliver;
```